keras_cv/models/backbones/detectron2/detectron2_backbone.py

```diff
@@ -0,0 +1,215 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+
+from keras_cv.api_export import keras_cv_export
+from keras_cv.backend import keras
+from keras_cv.layers.detectron2_layers import AddPositionalEmbedding
+from keras_cv.layers.detectron2_layers import ViTDetPatchingAndEmbedding
+from keras_cv.layers.detectron2_layers import WindowedTransformerEncoder
+from keras_cv.models import utils
+from keras_cv.models.backbones.backbone import Backbone
+from keras_cv.models.backbones.detectron2.detectron2_backbone_presets import (
+    backbone_presets,
+)
+from keras_cv.utils.python_utils import classproperty
+
+
+@keras_cv_export("keras_cv.models.ViTDetBackbone")
+class ViTDetBackbone(Backbone):
```

Comment:
Sorry if we've already been over this -- why is the name of the backbone different from the name of the file that it's in?

Should we rename the files `vit_det_backbone`?

Reply:
> why is the name of the backbone different from the name of the file that it's in?

I think `ViTDetBackbone` is more descriptive than `Detectron2Backbone`. I called the namespace `detectron2` since that's where the ViT model comes from.

> Should we rename the files `vit_det_backbone`?

I don't mind doing that. Will push this in the next batch of commits.

Reply:
Done.